qabelbox/src/test/java/de/qabel/qabelbox/storage/notifications/StorageNotificationTest.java

```diff
@@ -90,16 +90,18 @@ private void checkNotification(Queue<BoxUploadingFile> queue) {
         Notification notification = shadowManger.getAllNotifications().get(0);
         ShadowNotification shadowNotification = shadowOf(notification);
 
-        int expectedProgress = (currentUpload == null ? 100 : currentUpload.getUploadStatusPercent());
-        assertEquals(expectedProgress, shadowNotification.getProgressBar().getProgress());
+        if (currentUpload != null) {
```

Comment:
There shouldn't be if statements in tests. You can restructure the test so that you don't have an uncertainty on wether "currentUpload" may be null. The test can be split in multiple smaller tests in which you always know if "currentUpload" is null. Having such a large assertion method is a code smell.

One way to restructure the code is to extract creation of the android notification in another class "StorageNotificationSender" and test that separateley. When you then test the StorageNotificationManager, you can substitute the StorageNotification sender with a mock and don't have to test the inner working of an android notification every time. The checkNotification() assertion could then become very simple and easy to read.